components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,311 @@
+        Assert.assertEquals(action.getName(), result.getName());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getType(), result.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 5)
+    public void testUpdateAction() throws ActionMgtException, SQLException {
+
+        Action action = buildMockAction(true,true);
+        Action updatingAction = new Action.ActionRequestBuilder()
+                .name("Pre Issue Access Token")
+                .description("To configure pre issue access token")
+                .endpoint( new EndpointConfig.EndpointConfigBuilder()
+                        .uri("https://sample.com")
+                        .authentication(new AuthType.AuthTypeBuilder()
+                                .type(AuthType.AuthenticationType.BASIC)
+                                .properties(Arrays.asList(
+                                        new AuthProperty.AuthPropertyBuilder().name("username").value("updatingadmin").isConfidential(true).build(),
+                                        new AuthProperty.AuthPropertyBuilder().name("password").value("updatingadmin").isConfidential(true).build()))
+                                .build())
+                        .build())
+                .build();
+        Action result = daoImpl.updateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), updatingAction, action, TENANT_ID);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(updatingAction.getName(), result.getName());
+        Assert.assertEquals(updatingAction.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(updatingAction.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(
+                updatingAction.getEndpoint().getAuthentication().getType(), result.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 6)
+    public void testDeactivateAction() throws ActionMgtException, SQLException {
+
+        Action action = buildMockAction(true,true);
+        Assert.assertEquals(Action.Status.ACTIVE, action.getStatus());
+        Action deactivatedAction = daoImpl.deactivateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), TENANT_ID);
+        Assert.assertEquals(Action.Status.INACTIVE, deactivatedAction.getStatus());
+    }
+
+    @Test(priority = 7)
+    public void testActivateAction() throws ActionMgtException, SQLException {
+
+        Action action = buildMockAction(true,true);
+        Action deactivatedAction = daoImpl.deactivateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), TENANT_ID);
```

Comment:
Since we have deactivated the action in above method we dont need to deactivate the action again